### PR TITLE
Remove enable multi worker flag in gdk settings

### PR DESCRIPTION
**Pull request by jennifer-at-improbable-io**
#### Description
Remove the bEnableMultiWorker flag from GDKSettings. The SpatialNetDriver will now always create a LayeredLBStrategy, even if it just has one layer with a 1x1 grid. 

Some places where we used to check if bEnabledMultiWorker, we now just always execute. Others, we check for whether there are multiple layers enabled.

One breaking change is that I added a World parameter to SpatialStatics::IsOffloadingEnabled, because we don't know anymore until we've connected to the deployment. 

#### Release note
REQUIRED: Add a release note to the `##Unreleased` section of CHANGELOG.md. You can find guidance for writing useful release notes [here](../SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md). Documentation changes are exempt from this requirement.

#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?

STRONGLY SUGGESTED: How can this be verified by QA?

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 1]
```diff
@@ -90,9 +89,9 @@ void USpatialGDKSettings::PostInitProperties()
 
 	// Check any command line overrides for using QBI, Offloading (after reading the config value):
 	const TCHAR* CommandLine = FCommandLine::Get();
-	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideSpatialOffloading"), TEXT("Offloading"), bEnableMultiWorker);
+	// CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideSpatialOffloading"), TEXT("Offloading"), bEnableMultiWorker);
```
Comment (jennifer-at-improbable-io) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp:92]: What are the circumstances where we actually pass command line flags for these? Can I just remove them?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -435,7 +432,10 @@ void USpatialNetDriver::CreateAndInitializeLoadBalancingClasses()
 
 		if (WorldSettings == nullptr || WorldSettings->DefaultLayerLockingPolicy == nullptr)
 		{
-			UE_LOG(LogSpatialOSNetDriver, Error, TEXT("If EnableMultiWorker is set, there must be a Locking Policy set. Using default policy."));
+			if (WorldSettings->DefaultLayerLoadBalanceStrategy != nullptr)
+			{
+				UE_LOG(LogSpatialOSNetDriver, Error, TEXT("If a Load balancing strategy is set, there must be a Locking Policy set. Using default policy."));
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:437]: this checks load balancing strat but log refers to locking policy?

Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:437]: The locking policy is checked on 433. It's now valid to have neither LBStrategy nor Locking Policy set, but if you have a LB, you should also have a locking policy.

Comment (alastairdglennie) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:437]: ah yea, my bad

Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:437]: Although I do need a WorldSettings != nullptr check here.

Comment (alastairdglennie) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:437]: The WorldSEttings==nullptr conditional on the line above seems weird here, if that's true, we still immediately check WorldSettings->DefaultLayerLoadBalanceStrategy - should we have a separate early exit for that being nullptr?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 3]
```diff
@@ -832,16 +832,13 @@ bool USpatialSender::WillHaveAuthorityOverActor(AActor* TargetActor, Worker_Enti
...
+			if (AuthorityIntentComponent->VirtualWorkerId != NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId())
 			{
-				if (AuthorityIntentComponent->VirtualWorkerId != NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId())
-				{
-					WillHaveAuthorityOverActor = false;
-				}
+				WillHaveAuthorityOverActor = false;
 			}
 		}
 	}
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp:844]: I feel like this whole function might be redundant as of recently. Can we check authority role where this function is called instead> Checking intent this way seems like we might be overcomplicating it, authority role should encapsulate local changes to auth intent, and if it doesn't we might've done something wrong?

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 4]
```diff
@@ -89,9 +89,8 @@ void USpatialGDKSettings::PostInitProperties()
 
 	// Check any command line overrides for using QBI, Offloading (after reading the config value):
 	const TCHAR* CommandLine = FCommandLine::Get();
-	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideSpatialOffloading"), TEXT("Offloading"), bEnableMultiWorker);
 	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideHandover"), TEXT("Handover"), bEnableHandover);
-	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideLoadBalancer"), TEXT("Load balancer"), bEnableMultiWorker);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideLoadBalancing"), TEXT("Load balancing"), bOverrideLoadBalancing);
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp:93]: You'll need to update the NFR tests which pass in OverrideSpatialOffloading to the CLI

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp [hunk 5]
```diff
@@ -61,28 +61,17 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
...
-	const USpatialGDKSettings* SpatialSettings = GetDefault<USpatialGDKSettings>();
-
+	// Add Load Balancer Attribute. If this is a single worker deployment, this will be just be the single worker.
 	WorkerAttributeSet WorkerAttributeOrSpecificWorker = SpatialConstants::UnrealServerAttributeSet;
-	VirtualWorkerId IntendedVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
-
-	// Add Load Balancer Attribute if we are using the load balancer.
-	bool bEnableMultiWorker = SpatialSettings->bEnableMultiWorker;
-	if (bEnableMultiWorker)
+	VirtualWorkerId IntendedVirtualWorkerId = NetDriver->LoadBalanceStrategy->GetLocalVirtualWorkerId();
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp:66]:
```suggestion
	const VirtualWorkerId IntendedVirtualWorkerId = NetDriver->LoadBalanceStrategy->GetLocalVirtualWorkerId();
```

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 2 above)
Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:437]: Actually, I refactored this to make the logic simpler and clearer by using the bEnableMultiWorkerflag.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 3 above)
Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp:844]: So I put a break point in the false return and ran both our RPC unit tests and the only RPC gym we have, and neither hit the breakpoint. I spent a bit of time looking at the code, and I think you're right (and even if not, we just wouldn't drop a RPC, so the worst case is a performance drop) but I'm uncomfortable making this change right before 0.10. How would you feel about a follow-up ticket?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp (hunk 5 above)
Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp:66]: Done.

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp (hunk 4 above)
Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp:93]: I'll follow up with you offline on this.

Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp:92]: Done. I had been including it in the general "all the things changed, read the docs" CHANGELOG entry, but you're right it's probably better to call it out explicitly.

File: SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h [hunk 6]
```diff
@@ -308,4 +304,6 @@ class SPATIALGDK_API USpatialGDKSettings : public UObject
 	/** Experimental feature to use SpatialView layer when communicating with the Worker */
 	UPROPERTY(Config)
 	bool bUseSpatialView;
+
+	bool bOverrideLoadBalancing;
```
[the remark this answers is not in the document]
Comment (jennifer-at-improbable-io) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h:308]: done.

**Pull request merged**